src/main/java/fi/fmi/avi/model/immutable/PhenomenonGeometryWithHeightImpl.java

```diff
@@ -47,6 +48,10 @@ public static Builder from(final PhenomenonGeometryWithHeight value) {
                         .setUpperLimitOperator(value.getUpperLimitOperator())
                         .setGeometry(TacOrGeoGeometryImpl.immutableCopyOf(value.getGeometry()))
                         .setApproximateLocation(value.getApproximateLocation())
+                        .setMovingDirection(value.getMovingDirection())
+                        .setMovingSpeed(value.getMovingSpeed())
+                        .setIntensityChange(value.getIntensityChange())
+                        .setAnalysisType(value.getAnalysisType())
```

Comment:
`NoVaExpected` is not set? Anyway, as order of setXXX invocations is different from interface property order, this is a bit harder to check. I would suggest setting properties in same order than declared in the interface, superinterfaces first.

Reply:
Fixed